### PR TITLE
Using a memoized Empty Object for StyleTag rendererAttributes prop

### PR DESCRIPTION
**Pull request by RamyRafik**
## Previous Behavior
When a FluentProvider component re-renders, the associated Style element is removed and re-inserted to the HEAD. Aside from the Perf improvement, this can lead to rendering un-styled child Components, if the Browser gets few frames between the Deletion and Insertion of the Style element.

Example, taken from PowerApps bug (No PII, All is debug-data)
Resizing triggers re-rendering of React Tree, including FluentProvider

5336dad7-879b-47c0-bd90-2f3c480ec218

Adding a "Break on -> Node Removal" to the Style element revealed the cause.

The root cause was passing a new Empty Object in the Props to `useFluentProviderThemeStyleTag` every time the FluentProvider re-renders.

## New Behavior
We are now using a memoized empty Object, similar to the `EMPTY_OBJECT` which was created previously for the same issue.

## Related Issue(s)
N/A- There is an issue on Power Apps  product (Bug # 3824191) due to current behavior. I didn't create an issue on FluentUI repo since the fix is less time consuming.

### CODE REVIEW

File: packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const useFluentProvider_unstable = (
   const { styleTagId, rule } = useFluentProviderThemeStyleTag({
     theme: mergedTheme,
     targetDocument,
-    rendererAttributes: renderer.styleElementAttributes ?? {},
+    rendererAttributes: renderer.styleElementAttributes ?? EMPTY_OBJECT,
```
Comment (QuangThuanDinh) [inline packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts:68]: I think we should introduce a new empty_object, not reuse the current one to make sure we not sharing memory with other components

**Review by layershifter — APPROVED** [review]

**Comment by RamyRafik** [comment]
> @RamyRafik nice catch 👍 Please run `yarn change` to create a change file and then push it.

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Avatar-mount_base.html">683</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Avatar-mount.html">644</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Button-mount_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Button-mount.html">293</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Field-mount_base.html">1103</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Field-mount.html">1146</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProvider-mount_base.html">722</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProvider-mount.html">710</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-mount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-mount.html">84</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">73</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">32</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/MakeStyles-mount_base.html">855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/MakeStyles-mount.html">855</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Persona-mount_base.html">1832</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/Persona-mount.html">1748</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/SpinButton-mount_base.html">1432</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30960/perf-test-react-components/SpinButton-mount.html">1374</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**